src/_pytest/pytester.py

```diff
@@ -1278,7 +1279,9 @@ def spawn_pytest(
 
         """
         basetemp = self.tmpdir.mkdir("temp-pexpect")
-        invoke = " ".join(map(str, self._getpytestargs()))
+        if args is None:
```

Comment:
Building a command string by joining arguments like this is unsafe, should change `spawn` to accept args as well, and pass them to `pexpect.spawn` as separate arguments. According to the pexpect [doc](https://pexpect.readthedocs.io/en/stable/api/pexpect.html#pexpect.spawn.__init__) this is supported.

Backward compatibility might be a little tricky, but should be doable.